v8/payloads/template.ts

```diff
@@ -0,0 +1,23 @@
+/**
+ * Types extracted from https://github.com/discord/discord-api-docs/blob/master/docs/resources/Template.md
+ */
+
+import type { APIUser } from './user';
+import type { APIGuild } from './guild';
+
+/**
+ * https://github.com/discord/discord-api-docs/blob/master/docs/resources/Template.md#template-structure
+ */
+export interface APITemplate {
+	code: string;
+	name: string;
+	description: string | null;
+	usage_count: number;
+	creator_id: string;
+	creator: APIUser;
+	created_at: string;
+	updated_at: string;
+	source_guild_id: string;
+	serialized_source_guild: Partial<APIGuild>;
```

Comment:
Apparently..this is the same as the `RESTPostAPIGuildCreateJSONBody`?

Reply:
no? serialized_source_guild has a lot more than those 2 fields ([example](https://github.com/discord/discord-api-docs/blob/master/docs/resources/Template.md#example-template-object))

Reply:
Which are also present in `RESTPostAPIGuildCreateJSONBody`, which has a LOT of fields (like roles, channels, etc...) and it was mentioned in some of the PRs you linked

Reply:
See https://github.com/discord/discord-api-docs/pull/2144#issuecomment-708490998

Reply:
oh sorry i misread, so use that but replace icon with icon_hash? looks like it also has a `description` field though

Reply:
Is..the JSON body for create guild outdated? Or are there values only for templates? 🤔

Reply:
`icon_hash` is an exception apparently

`description` isn't really useful in creating guilds as it's only for discoverable (or preview_enabled) guilds

Reply:
Useful or not, we should document it if it can be used 👍

Reply:
yeah so it should stay APIGuild, since that's the one with the description property and that's what it's documented as

Reply:
Yeah..I don't know what to say about that really.. Feedback needed for this one.

Discord claims `serialized_source_guild` is exactly the data that the `Create Guild` endpoint takes (in this case `RESTPostAPIGuildCreateJSONBody` which also accounts for IDs being numbers (!)). Per https://github.com/discord/discord-api-docs/pull/2144#issuecomment-708490998, seems like icon_hash isn't even used anymore (although it's still set... neat). While typing this as `Partial<APIGuild>` isn't necessarily *wrong*, it's also not *right*...

CC @kyranet @iCrawl what do you say?

Reply:
We can still make a `APITemplateSerializedSourceGuildResult`, and if the fields are more similar to `RESTPostAPIGuildCreateJSONBody` but the properties differ, we can omit the ones missing and add the ones that aren't.

Reply:
As it stands, `RESTPostAPIGuildCreateJSONBody` is most accurate.. a small `type APITemplateSerializedSourceGuild = RESTPostAPIGuildCreateJSONBody & { whateverIsMissing }` is plenty

Reply:
@advaith1 ^ since your PR(s?) were merged